### DISCUSSION TITLE
sampling_voxel vs sampling_poisson

### DISCUSSION THREAD

**Discussion by wiesehahn** [r-lidar/lasR#85] [Q&A]
I am wondering if there should be some difference between the sampling strategies? As I tunderstood voxel takes a random point within that voxel while poisson assures minimim distance.

The following looks like the same points get sampled, but I think I just miss something here. 
.
``` r
library(lasR)
f <- system.file("extdata", "Topography.las", package="lasR")

read <- reader_las()
vox <- sampling_voxel(10)
write <- write_las(ofile = templas())
pipeline <- read + vox + write
ans <- exec(pipeline, on = f)
las <- lidR::readLAS(ans)

read <- reader_las()
poi <- sampling_poisson(10)
write <- write_las(ofile = templas())
pipeline <- read + poi + write
ans <- exec(pipeline, on = f)
las2 <- lidR::readLAS(ans)

summary(las@data$gpstime)
#>      Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
#> 220367381 220367382 220367383 220367383 220367384 220367385
summary(las2@data$gpstime)
#>      Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
#> 220367381 220367382 220367383 220367383 220367384 220367385
```

**Comment by wiesehahn** [comment]
sorry, gpstime was just the wrong attribute to check here. actually there is a difference.

**Comment by Jean-Romain** [comment]
- voxel sampling: 1 random point per voxel. It is (relatively) quick and requires little memory. 
- poisson sampling: is [poisson sampling](https://en.wikipedia.org/wiki/Poisson_sampling). It produces the best sampling I know (very homogeneous, preserves the maximum amount of details) but is more computationally demanding and uses more memory.

At first glance they look a little similar but if you look carefully Poisson sampling is better. Well, better for what? I guess we can find useful application for voxel sampling too.

The fact that your two summaries are identical is due to (1) gpstime is not the appropriate metric (2) number of digits printed (3) 10 is relatively big. Test 5 and 3.5 respectively to get identical densities and plot the point clouds.
